### DISCUSSION TITLE
Language intelligence inside HTML template strings with latest release?

### DISCUSSION THREAD

**Discussion by aral** [helix-editor/helix#8842] [Q&A] [answered]
I read that the latest release has support for multiple LSPs per page so I was wondering if something like this is now possible:

```js
const html = html`
  <h1>This has language intelligence</h1>
`
```

Currently, the Typescript LSP is intelligent enough to highlight code in html (and css) template strings but you don’t get any language intelligence for HTML in them.

Similarly, for HTML in Markdown, etc.

Any thoughts on what, if anything, I can do to configure Helix to get this working?

Also, is anyone else using Helix for web development? Is it worth maintaining a configuration for web development that does the common things out of the box?

**Comment by pascalkuthe** [comment, marked as answer]
this is not possible. Running language servers on subranges of a document is not supported by the LSP standard.

**Comment by aral** [comment]
Ah, well, all the more reason to stay with your inline diagnostics branch, then – thanks, Pascal ;)
